git/packet_reader.go

```diff
@@ -0,0 +1,62 @@
+package git
+
+import (
+	"io"
+	"math"
+)
+
+type packetReader struct {
+	proto *protocol
+
+	buf []byte
+}
+
+var _ io.Reader = new(packetReader)
+
+func (r *packetReader) Read(p []byte) (int, error) {
+	var n int
+
+	if len(r.buf) > 0 {
+		// If there is data in the buffer, shift as much out of it and
+		// into the given "p" as we can.
+		n = int(math.Min(float64(len(p)), float64(len(r.buf))))
```

Comment:
`math.Min` accepts two float64s as its arguments, and the `len` builtin returns an `int`, so the cast must be preformed 😭